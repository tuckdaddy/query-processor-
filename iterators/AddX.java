package iterators;


import java.util.Iterator;

public class AddX implements Iterator{
      
    // The Iterator that this apply  object will get its input from
    private final Iterator<Integer> input;	
    private final int x;
    // Constructs an instance of Addx with and int and an iterator
    public AddX(int x, Iterator<Integer> input) {
        this.input = input;
        this.x = x;
    }

    @Override
    public boolean hasNext() {
        return input.hasNext();
    }

    @Override
    // Method that overrides Iterators next() that adds the parameter x
    // to the data in input.next()
    public Integer next() {
        if (!hasNext()) throw new IllegalStateException();
        return input.next() + x;
    }
}       
