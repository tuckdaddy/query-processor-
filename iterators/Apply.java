package iterators;


import java.util.Iterator;

public class Apply<InT,OutT> implements Iterator<OutT> {
    // The function that will be applied to each input element to make an output element
    //Higher order function 
    // InT indicates input data, OutT indicates output data
    private final ApplyFunction<InT,OutT> f;

    // The Iterator that this Apply object will get its input from
    private final Iterator<InT> input;

    public Apply(ApplyFunction<InT, OutT> f, Iterator<InT> input) {
        this.f = f;
        this.input = input;
    }

    @Override
    public boolean hasNext() {
        return input.hasNext();
    }

    @Override
    public OutT next() {
        if((!hasNext())) throw new IllegalStateException();
        return f.apply(input.next());
    }
}
