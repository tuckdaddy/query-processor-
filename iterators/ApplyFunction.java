package iterators;

// Interface for classes to define a function from InT to OutT
public interface ApplyFunction<InT, OutT> {
		public OutT apply(InT x);
}
