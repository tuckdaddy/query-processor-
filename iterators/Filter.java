package iterators;

import java.util.Iterator;
import java.util.LinkedList;
import java.util.List;
import java.lang.NullPointerException;

// Iterator that uses a Predicate to filter out elements from the input
public class Filter<T> extends FlatApply<T, T> {

    public Filter(Predicate<T> p, Iterator<T> input) {
        // you DO NOT need to modify the constructor
        super(new FilteringFlatApplyFunction<>(p), input);
    }

    // uses a Predicate to decide whether the input element is output or not
    private static class FilteringFlatApplyFunction<T> implements FlatApplyFunction<T, T> {
        private Predicate<T> derp; 
        public FilteringFlatApplyFunction(Predicate<T> p){
            this.derp = p;
        }
        
        // if x passes the predicate, add x to the new list 
        @Override 
        public List<T> apply(T x){
            List <T> lst = new LinkedList<>();
            boolean res = derp.check(x);
            if (res == true){
                lst.add(x);
                return lst;
            }
            else {
                return lst;
            }
        }
    }

    // You DO NOT need to define hasNext() and next(). FlatApply provides them already.
}