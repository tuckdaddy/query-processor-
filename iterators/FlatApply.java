package iterators;

import java.util.Queue;
import java.util.LinkedList;
import java.util.List;
import java.util.Iterator;

// Iterator that may produce 0 or more output elements for every input element
public class FlatApply<InT,OutT> implements Iterator<OutT> {
    private final Iterator <InT> input;
    private final FlatApplyFunction<InT,OutT> f;
    private final Queue<OutT> q;
    
    // Constructs a new linked list queue everytime constructor is called
    public FlatApply(FlatApplyFunction<InT,OutT> f, Iterator<InT> input) {		
        this.input = input;
        this.f = f;
        this.q = new LinkedList<>();
    }

    @Override
    // If the queue is not empty return true, else fill the queue by applying the 
    //  function until input.hasNext() is false
    //  but there is a check again if the queue is empty to eliminate any errors
    public boolean hasNext() {
        if (!q.isEmpty()){return true;}
        else{
            while(q.isEmpty() && input.hasNext()){
                List<OutT> res = f.apply(input.next());
                q.addAll(res);
            }
            if(q.isEmpty()) return false;
            return true;
        }
    }

    @Override
    public OutT next() {
        if((!hasNext())) throw new IllegalStateException();
        return q.poll();
    }
}
