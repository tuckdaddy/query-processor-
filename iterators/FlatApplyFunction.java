package iterators;


import java.util.List;

// Interface for classes to define a function from InT to List<OutT>
public interface FlatApplyFunction<InT, OutT> {
		public List<OutT> apply(InT x);
}
