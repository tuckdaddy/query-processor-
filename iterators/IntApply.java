package iterators;


import java.util.Iterator;

public class IntApply implements Iterator {
// The function that will be applied to each input element to make an output element
    private final IntApplyFunction f;

// The Iterator that this Apply object will get its input from
    private final Iterator<Integer> input;		
    // Costructor that takes a function as input and runs it through a function, the value is the output
    // and an input iterator
    // Just like ApplyX but instead of passing a variable you pass a function
    public IntApply(IntApplyFunction f, Iterator<Integer> input) {
        this.input = input;
        this.f = f;
    }

    @Override
    public boolean hasNext() {
        return input.hasNext();
    }

    @Override
    // f is an obect of IntApplyFunction allowing access to its method apply
    // the data passes to apply is input.next()
    public Integer next() {
        if (!hasNext()) throw new IllegalStateException();
        return(f.apply(input.next()));
    }
}
