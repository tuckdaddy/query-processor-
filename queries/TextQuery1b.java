package queries;


import iterators.Apply;
import iterators.ApplyFunction;
import java.util.Iterator;
import readers.TextFileReader;

// return the 12th word in every file
public class TextQuery1b {
    public static void main(String[] args) {
	Iterator<Pair<String,String>> filenameAndContents = new TextFileReader("../sci.space");
        Iterator<String> contents = new Apply(new TakeRight<>(), filenameAndContents);
        Iterator<String> TwelthWords = new Apply(new TwelfthWord(), contents);
        // prints evrery 12th word in each file
        while (TwelthWords.hasNext()) {
            System.out.println(TwelthWords.next());
        }
    }	

    // The only change you should make to this file:
    // define TwelfthWord class here
    // Its apply function should take a String and return 
    // only the 12th word as a String (where words are defined as separated by a " ")
    // Any class that "implements"	
    // ApplyFunction must provide an implementation of apply()
    private static class TwelfthWord implements ApplyFunction<String,String>{
        @Override
        // takes a sting as a parameter, and returns a string as output
        // in  this case the twelfth word
        public String apply(String x){
            String s[] = x.split(" ");
            return s[11];
        }
    }

    private static class TakeRight<L,R> implements ApplyFunction<Pair<L,R>, R> {
        @Override
        public R apply(Pair<L, R> x) {
            return x.right;	
        }
    }
}
