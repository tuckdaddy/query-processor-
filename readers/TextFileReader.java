package readers;


import queries.Pair;
import java.io.BufferedReader;
import java.io.File;
import java.io.FileReader;
import java.io.IOException;
import java.util.Iterator;

public class TextFileReader implements Iterator<Pair<String,String>> {
		private final File[] files;
		private int curFile;

		public TextFileReader(String folderPath) {
			File folder = new File(folderPath);
			this.files = folder.listFiles();
			this.curFile = 0;
		}
		
		@Override
		public boolean hasNext() {
			return curFile < files.length;
		}

		/* returns a Pair containing (filename, contents), i.e.,
		   Pair.left=filename and Pair.right=contents
		*/
		@Override
		public Pair<String, String> next() {
			try {
				String name = files[curFile].getName();
				BufferedReader in = new BufferedReader(new FileReader(files[curFile]));
				StringBuilder sb = new StringBuilder();
				String s = in.readLine();
				while (s != null) {
					sb.append(s);
					s = in.readLine();
				}
				curFile++;
				return new Pair(name, sb.toString());
			} catch (IOException ex) {
				throw new RuntimeException(ex);
			}
		}
}
