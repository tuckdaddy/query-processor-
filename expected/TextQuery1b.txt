Taylor)Subject:
Space
FAQ
FAQ
FAQ
Frequently
-
-
-
-
FAQ
FAQ
FAQ
Space
FAQ
FAQ
Space
Astro/Space
Re:
sci.space:60152
Apr
neededMessage-ID:
Apr
DigestSender:
5
MIT
guy
Office
6
Forwarded
at
<C51FK8.M4I.1@cs.cmu.edu>X-Added:
not
Re:
Celebrate
do
Space
if
Apr
OffMessage-ID:
Griffin
6
<1993Apr5.191701.1@aurora.alaska.edu>Date:
be
Apr
Apr
/
TestMessage-ID:
6
Re:
Re:
what
Magellan
sci.space:60186
automate
6
Space
(was
Space
93
EnglishMessage-ID:
gov
What
<1993Apr6.125608.7506@mksol.dseg.ti.com>Organization:
<1993Apr6.130112.7692@mksol.dseg.ti.com>Organization:
9Sender:
<1993Apr6.142950.26579@fys.ruu.nl>Summary:
at
Comet
93
<1pscc6INNebg@news.aero.org>Date:
Apr
Apr
foodLines:
envelopeMessage-ID:
Comet
(news
if
Apr
news@cs.ucf.edu
93
<1993Apr6.121843.1@max.u.washington.edu>Date:
Apr
Shafer)Date:
Apr
in
Apr
with
13:56:53
Apr
1993
1993
Nears
and
Comet
1993
GuidanceMessage-ID:
prb@access.digex.com's
Re:
news+@cs.cmu.eduOrganization:
SSF
NamedMessage-ID:
15:50:07
93
of
(was
(was
Astronaut)Message-ID:
VNEWS
by
<1993Apr6.210138.1857@mksol.dseg.ti.com>Lines:
<1993Apr6.210827.2383@mksol.dseg.ti.com>Organization:
<1psogpINNksq@rave.larc.nasa.gov>Date:
Fidonet.
easy
station,
<C5329M.By@csc.ti.com>Sender:
and
dpage@ra
Forwarded
7
Spock
Apr
Recent
Sail
AvailableDate:
Re:
15
<1qkmkiINNep3@mojo.eng.umd.edu>Date:
No,
by
sci.space,sci.astro,alt.sci.planetarySubject:
jmcocker@c00068-100lez.eos.ncsu.eduSender:
Re:
Update
Flight
Operation
<1993Apr15.204210.26022@mksol.dseg.ti.com>Lines:
Vulcan?
Temporary
Space
Evil
DigestSender:
<C5K41u.Hnx.1@cs.cmu.edu>X-Added:
sci.space?Message-ID:
VAX/VMS
GMTSender:
works?Message-ID:
Fri,
<giese.734934155@katarina.dev.cdx.mot.com>Sender:
Science
16
09:49:09
<3918@eagle.ukc.ac.uk>Date:
Inc.Date:
news@aio.jsc.nasa.gov
Public
Fri,
the
<H0Z32B3w165w@inqmind.bison.mb.ca>Date:
Astronaut)Date:
How
Fri,
AT&T
16
<C5LEHy.KB0.1@cs.cmu.edu>X-Added:
<2BCEDFE2.18001@news.service.uci.edu>Newsgroups:
usenet@convex.com
works?Sender:
works?Message-ID:
Apr
US-Russian
japanese
drive
netnewsMessage-ID:
Fri,
drive
16
Apr
16
16
Feb
1993
<1993Apr16.233403.12924@gmuvax2.gmu.edu>Organization:
(News
<pgf.735012282@srl03.cacs.usl.edu>Sender:
Sat,
Space
Gun,
Gun,
floorMessage-ID:
of
drive
04:55:31
93
News
Apr
well.sf.ca.usOrganization:
Apr
Orion
Space
Space
EDTFrom:
17
Sat,
CDTOrganization:
1993
17
ShuttleMessage-ID:
drive
17
Mystery
drive
Evil
23
23
<4617416f8@ofa123.fidonet.org>Date:
SSTO
<degroffC5o2IK.99v@netcom.com>Organization:
WANTED
Communications
had
(USENET
<C5owCB.n3p@world.std.com>Organization:
Re:
Space
RELEASE
Account)Message-ID:
<1993Apr18.224248.29493@aio.jsc.nasa.gov>Sender:
Smithsonian
93
Clementine?)Message-ID:
Comet
(was
to
93
<uckC5q1so.G2p@netcom.com>Organization:
<1993Apr18.020535.28971@mnemosyne.cs.du.edu>Sender:
drive
Temporary
Evil
CDTOrganization:
BSTNewsgroups:
BSTNewsgroups:
Re:
(Uutis
had
message
first
Orion
thread
drive
May
Space
Update
ChicagoOrganization:
Virginia's
<1993Apr19.213334.11471@vdoe386.vak12ed.edu>Organization:
first
first
DebrisMessage-ID:
Manned
Exploration.
<C5r2I1.793@skates.gsfc.nasa.gov>Sender:
Federal
billion
at
Manned
article
Space
for
news@csus.eduOrganization:
$1
Exploration.
23:39:07
ADDRESSES
first
sonic
Magellan
peripherals)Newsgroups:
peripherals)Newsgroups:
Mission
news@raven.alaska.edu
BSTNewsgroups:
$1
<1993Apr20.020259.1@aurora.alaska.edu>Lines:
without
EDTFrom:
Evil
DC-X
news@fs7.ece.cmu.edu
message
Space
Magellan
Manned
(Merlin
SailMessage-ID:
billion
<C5ssxD.1u5.1@cs.cmu.edu>X-Added:
had
had
DigestSender:
DigestSender:
DigestSender:
drive
$6
Exploration.
20
DataDate:
Keeping
moonMessage-ID:
Keeping
Orion
Orbit
<1993Apr20.211638.168730@zeus.calpoly.edu>Date:
the
the
gifs
the
$1
sky.Message-ID:
-
21
Apr
usenet@mnemosyne.cs.du.edu
first
Spacecraft
Apr
billion
21
sunset
the
<1993Apr20.234427.1@aurora.alaska.edu>Date:
peripherals)Newsgroups:
Apr
billion
Apr
the
<1993Apr20.140123.13653@kocrsv01.delcoelect.com>Sender:
the
Apr
sci.space,sci.astro,alt.sci.planetarySubject:
Beacons!Message-ID:
the
1993
Apr
Evil
of
By
<1993Apr21.134436.26140@mksol.dseg.ti.com>Organization:
Lunar
$1
not
Wed,
Apr
2005
Wed,
the
21
18:23:05
(was:Why
Apr
Wed,
Keeping
93
to
21
Mile-Long
billion
Comet
EclipseDate:
news@nas.nasa.gov
93
Keeping
had
MotorolaDate:
Account)Message-ID:
Proton/Centaur?Message-ID:
Re:
Re:
the
Mars
the
(was:Why
(USENET
32Sender:
Apr
93
1993
Apr
moonDate:
Apr
<C5vzrv.C29@fs7.ece.cmu.edu>Sender:
Re:
billion
Vandalizing
21
Apr
STILL:
Mining.Message-ID:
Comet
Re:
why?Message-ID:
Thu,
Nikan
Sunrise/
SOME
Spacecraft
Apr
of
Keeping
Generation
Mining.Message-ID:
Apr
be
message
found?Message-ID:
by
Space
billion
Astronaut)Message-ID:
<1993Apr22.170931.5037@lambda.msfc.nasa.gov>Organization:
Re:
on
22
give
(news
Re:
CDTNewsgroups:
1993
nsmca@aurora.alaska.edu's
University
read
ORBIT
1993
(was:Why
why?Sender:
way
MotorolaDate:
Re:
sci.space,sci.astro,alt.aci.planetary,alt.astrologySubject:
newtout
22
$1
BY
Update
Inflatable
Comet
PLANETS
the
21
<C5wyn4.G4v@news.Hawaii.Edu>Sender:
23
why?Message-ID:
23
Sunrise/
NAvigation
1993
23
give
Apr
System)Nntp-Posting-Host:
1993
<C5xp0K.G79@brunel.ac.uk>Organization:
news@cs.ucf.edu
Space
Loss
Why
Why
Re:
Resigns
LINK!
(was
Command
<C5y0Kw.Mx0@zoo.toronto.edu>Date:
Keeping
Command
Feb
give
news@javelin.sim.es.comOrganization:
Loss
Servicing
<1r9666$jta@access.digex.net>Date:
23
STILL:
will
575-3539)Subject:
found?Organization:
Vandalizing
is
Hawking
Sunrise/
Project
Keeping
23
reminder
Re:
23
usenet@aio.jsc.nasa.gov
23
News
23
23
Feb
Apr
Observer
Update
found?Organization:
Chief
Fri,
Apr
15
20:47:22
thacker@rhea.arc.ab.caDate:
JSC
<1993Apr23.175120.1@stsci.edu>Date:
billion
STILL:
Vandalizing
(was
93
Apr
24
JSC
TrekMessage-ID:
Re:

Hawking
23
00:10:21
waves,
Gamma
Sat,
Apr
the
U)Subject:
free
California
landis@stsci.eduOrganization:
news@stsci.eduReply-To:
Alternative
Alternative
24
PLANETS
93
Where
Servicing
(USENET
launch
HST
IMAGES
the
Hawking
Apr
sunset
Who
23
Feb
Feb
"GLOBE",
cheap.Message-ID:
aurora.1993Apr25.151108.1Sender:
<1993Apr25.154449.1@aurora.alaska.edu>Date:
Ideas)Date:
Ray
<1469100033@igc.apc.org>Sender:
restore
(Rui
sky.Date:
2)Date:
26
Mission?Sender:
Sunrise/
25
25
Apr
Exploration.
26
Apr
2)Message-ID:
Gamma
CDTNewsgroups:
Gamma
JSC
MissionMessage-ID:
surveillance?Date:
Vandalizing
Vandalizing
Apr
Spacecraft
Forwarded
CD
Client)Organization:
93
<1993Apr26.191442.26802@butch.lmsc.lockheed.com>Sender:
<C63vvG.4J9@zoo.toronto.edu>Date:
Boom!
DigestSender:
(was
Command
"GLOBE",

Klaes)Subject:
Command
<1993Apr26.215851.26175@den.mmc.com>Date:
Redesign,
news@zeus.usq.edu.au
University
Ray
Mon,
by
are
<C646J3.3x3@news.Hawaii.Edu>Sender:
Apr
Space
Space
energetic
Gamma
Gamma
Vandalizing
Gamma
newtout
93
<1993Apr26.222659.1@aurora.alaska.edu>Lines:
"GLOBE",
Space
Tue,
Apr
Forwarded
<C65GLF.8Gx.1@cs.cmu.edu>X-Added:
HST
Space
moon
(was
BSTNewsgroups:
27
Gamma
Ray
<C659w7.IyD@fs7.ece.cmu.edu>Sender:
sci.physics,sci.astro,sci.space,sci.spaceSubject:
sci.physics,sci.astro,sci.space,sci.spaceSubject:
habitableMessage-ID:
27
27
give
sci.physics,sci.astro,sci.space,sci.spaceSubject:
sci.physics,sci.astro,sci.space,sci.spaceSubject:
Apr
Vandalizing
HST
<1993Apr27.121804.1@stsci.edu>Lines:
Re:
HST
and
surveillance?Message-ID:
DigestSender:
the
surveillance?Message-ID:
DigestSender:
news@cs.ucf.edu
Ray
the
SSTO
of
Mission
Space
cheap.X-Sender:
Tue,
<1993Apr27.174622.1@stsci.edu>Lines:
27
Space
moon
<1993Apr23.124759.1@fnalf.fnal.gov>Sender:
news@nimasterSender:
<C663u5.IKC@zoo.toronto.edu>Date:
27
Re:
evening
Wed,
Evening
of
28
term
Ray
20:08:41
temperature
Bursters

Gamma
sky.Message-ID:
Ray
stuff.Message-ID:
HST
Ray
New
temperature
of
term
Ray
Apr
Vandalizing
news+@cs.cmu.eduOrganization:
news+@cs.cmu.eduOrganization:
Ray
Apr
<1993Apr29.122312.15797@cs.rochester.edu>Date:
the
MoonMessage-ID:
Long
list
list
HST
MissionMessage-ID:
<1993Apr29.100808.1@stsci.edu>Lines:
moon
Apr
28
spaceDate:
15:49:31
<1993Apr28.143916.3903@aio.jsc.nasa.gov>Sender:
Higgins--
1993
the
of
of
dark
Ray
Gamma
Vandalizing
weightlessness
MissionMessage-ID:
575-3539)Subject:
and
WHere
GMTReferences:
are
spaceDate:
news@netnews.louisville.edu
Gamma
are
HST
Ray
does
Gamma
Gamma
HST
sci.spaceSubject:
Exploration
the
29
Ray
the
Space
of
Ray
the
Observer
Apr
Gamma
Command
HST
Re:
_is_
Long
Command
Update
WHere
93
Gamma
Gamma
Space
to
to
HST
NEwsgroup?Message-ID:
Imperial
29
<1993Apr29.162132.28366@hemlock.cray.com>Originator:
<OREILLY.93Apr29072526@olivia.la.asu.edu>Sender:
Forwarded
29
29
1993
Mission
System
mission_)Date:
04/29/93Message-ID:
HST
Green)Subject:
mission_)Message-ID:
News
30
Gamma
Long
news@raven.alaska.edu
Servicing
Bursters
(USENET
(Usenet
GMessage-ID:
Boldly?Message-ID:
Mission
Imperial
30
makes
InvestmentMessage-ID:
Servicing
<C6Az8z.pD@zoo.toronto.edu>Date:
Servicing

OK?Keywords:
30
Fri,
Message-ID:
HST
HST
of
the
Turner)Subject:
deployed
MissionMessage-ID:
Re:
Alternative
<1993Apr30.165603.573@mksol.dseg.ti.com>Organization:
Re:
Re:
575-3539)Subject:
Pluto
<152941830@ofa123.fidonet.org>Date:
Boeing)X-Sender:
Mission
Mission
Space
Re:
Space
HST
<1993Apr30.160814.1@stsci.edu>Lines:
sci.spaceDate:
Re:
Alternative
the
HST
Observer
<1993Apr30.203330.8893@mksol.dseg.ti.com>Lines:
30
Re:
Boldly?Message-ID:
Servicing
Marco?
Space
WHere
and
O'Shaughnessy)Subject:
Gamma
1
(was
Boldly?Message-ID:
news+@cs.cmu.eduOrganization:
news+@cs.cmu.eduOrganization:
Space
Hall
News
News
<1993May1.044441.1@aurora.alaska.edu>Lines:
Corp.Date:
27Sender:
Re:
cutting
ASTRO
temperature
Words
WHere
1
from
NASA/JSC/GM2,
23
feel
Space
(Re:
Deployable
Servicing
May
(Re:
servicing
Sat,
field
Re:
1993
cutting
HST
Ray
cutting
field
WHere
HST
and
<C6EHK8.1p9@brunel.ac.uk>Organization:
Friedman/Planetary
news@ucsu.Colorado.EDU
May
cutting
the
and
U.S.Date:
HST
May
servicing
1993
of
Rubin)Subject:
MissionMessage-ID:
21:24:05
HST
PSTOrganization:
1993
Re:
Vandalizing
from
the
criticsMessage-ID:
there
InvestmentMessage-ID:
Tue,
on
DC-X
(NN
May
Vancouver
(Was
<1993May12.092027.1765@noao.edu>Keywords:
McElwaine
Bizarre?
Thu,
is
ComparisonsMessage-ID:
anyway?Message-ID:
University
for
Forwarded
ComparisonsMessage-ID:
Another
Spencer
15
anyway?Message-ID:
on
May
Re:
anyway?In-Reply-To:
22:44:07
22:26:00
22:27:02
Forwarded
misc.headlines:42199Newsgroups:
misc.headlines:42200Path:
misc.headlines:42201Newsgroups:
misc.headlines:42207Newsgroups:
misc.headlines:42208Path:
U.S.
on
<1993May16.131644.5202@ke4zv.uucp>Reply-To:
Re:
Shuttle
10:56:40
May
misc.headlines:42220Newsgroups:
Mission
misc.headlines:42221Newsgroups:
anyway?Date:
misc.headlines:42224Newsgroups:
McKay)Subject:
misc.headlines:42225Path:
G.
Astrophysical
May
Coffman)Subject:
misc.headlines:42238Newsgroups:
misc.headlines:42239Newsgroups:
misc.headlines:42240Path:
misc.headlines:42241Newsgroups:
misc.headlines:42250Newsgroups:
misc.headlines:42251Newsgroups:
misc.headlines:42252Newsgroups:
would
would
misc.headlines:42254Newsgroups:
misc.headlines:42255Newsgroups:
be
misc.headlines:42266Path:
Lazio)Newsgroups:
criticsMessage-ID:
WEIGHTLESSNESS
misc.headlines:42267
postman@alw.nih.gov
misc.headlines:42268
sci.environment,misc.consumers,misc.invest,sci.astro,talk.environment,talk.politics/space,sci.space,rec.backcountry,misc.rural,misc.headlines,k12.chat.teacherPath:
May
sci.spaceDate:
news+@cs.cmu.eduOrganization:
(Chris
misc.headlines:42335Path:
